client/dashboard/profile-wizard/cart-modal.js

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * WooCommerce dependencies
+ */
+import { getSetting } from '@woocommerce/wc-admin-settings';
+import { getNewPath } from '@woocommerce/navigation';
+import { List } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import withSelect from 'wc-api/with-select';
+import { getProductIdsForCart } from 'dashboard/utils';
+import sanitizeHTML from 'lib/sanitize-html';
+
+class CartModal extends Component {
+	componentDidMount() {
+		document.body.classList.add( 'woocommerce-admin-full-screen' );
+		document.body.classList.add( 'woocommerce-profile-wizard__body' );
+	}
+
+	componentWillUnmount() {
+		document.body.classList.remove( 'woocommerce-admin-full-screen' );
+		document.body.classList.remove( 'woocommerce-profile-wizard__body' );
+	}
+
+	redirectToCart() {
+		const { productIds } = this.props;
+		const backPath = getNewPath( {}, '/', {} );
+		const { connectNonce } = getSetting( 'onboarding', {} );
+
+		if ( ! productIds.length ) {
+			return;
+		}
+
+		const url = addQueryArgs( 'https://woocommerce.com/cart', {
+			'wccom-site': getSetting( 'siteUrl' ),
+			'wccom-woo-version': getSetting( 'wcVersion' ),
+			'wccom-replace-with': productIds.join( ',' ),
+			'wccom-connect-nonce': connectNonce,
+			'wccom-back': backPath,
+		} );
+		window.location = url;
+	}
+
+	renderProducts() {
+		const { productIds } = this.props;
+		const { productTypes = {}, themes = [] } = getSetting( 'onboarding', {} );
+		const listItems = [];
+
+		productIds.forEach( productId => {
+			const productInfo = find( productTypes, productType => {
+				return productType.product === productId;
+			} );
+
+			if ( productInfo ) {
+				listItems.push( {
+					title: productInfo.label,
+					content: productInfo.description,
+				} );
+			}
+
+			const themeInfo = find( themes, theme => {
+				return theme.id === productId;
+			} );
+
+			if ( themeInfo && themeInfo.slug !== 'Storefront' ) {
+				listItems.push( {
+					title: sprintf(
+						__( '%s — %s per year', 'woocommerce-admin' ),
+						themeInfo.title,
+						decodeEntities( themeInfo.price )
+					),
+					content: <span dangerouslySetInnerHTML={ sanitizeHTML( themeInfo.excerpt ) } />,
+				} );
+			}
+		} );
+
+		return <List items={ listItems } />;
+	}
+
+	render() {
+		return (
+			<Modal
+				title={ __(
+					'Would you like to purchase and install the following features now?',
+					'woocommerce-admin'
+				) }
+				onRequestClose={ () => this.props.onClose() }
+				className="woocommerce-cart-modal"
+			>
+				{ this.renderProducts() }
+
+				<p>
+					{ __(
+						"You won't have access to this functionality until the extensions have been purchased and installed.",
+						'woocommerce-admin'
+					) }
+				</p>
+
+				<div className="woocommerce-cart-modal__actions">
+					<Button isLink onClick={ () => this.props.onClose() }>
+						{ __( "I'll do it later", 'woocommerce-admin' ) }
+					</Button>
+
+					<Button isPrimary isDefault onClick={ () => this.redirectToCart() }>
```

Comment:
Optional: might be a nice addition for slower connections if we add a loading state when this is clicked.

Reply:
Added in https://github.com/woocommerce/woocommerce-admin/pull/3444/commits/d7762abe8ad6717ef3db01792f1b1fe9f3584906.